### PR TITLE
Fix Windows build with posix specific stat APIs

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
@swift-ci please smoke test

File: include/llbuild/Basic/Stat.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#define S_ISFIFO(mode) ((mode) & _S_IFMT) == S_IFIFO
+#endif
+
+#if !defined(S_ISSOCK)
+#define S_ISSOCK(mode) ((mode) & _S_IFMT) == S_IFSOCK
+#endif
+
+#if !defined(_S_ISLINK)
+#define S_ISLNK(mode) ((mode) & _S_IFMT) == S_IFLNK
+#endif
```
Comment (ddunbar) [inline include/llbuild/Basic/Stat.h]: This isn't really the ideal fix, we should hide all of this stuff outside the header, but I'm fine with it for now to make forward progress on the Windows port.

**Comment by ddunbar** [comment]
@hughbe failures look related:
```
FAILED: /usr/bin/clang++    -I/home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/include -std=c++14 -fno-rtti -fno-exceptions -Wmost -Wdocumentation  -Woverloaded-virtual -Wparentheses -Wswitch -Wunused-function -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wsign-compare -Wnewline-eof -Wdeprecated-declarations -Winvalid-offsetof -Wnon-virtual-dtor -fPIC -include "/home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/include/libstdc++14-workaround.h" -Wdocumentation -Wimplicit-fallthrough -Wunreachable-code -Woverloaded-virtual -g -MMD -MT lib/Basic/CMakeFiles/llbuildBasic.dir/FileSystem.cpp.o -MF lib/Basic/CMakeFiles/llbuildBasic.dir/FileSystem.cpp.o.d -o lib/Basic/CMakeFiles/llbuildBasic.dir/FileSystem.cpp.o -c /home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/lib/Basic/FileSystem.cpp
In file included from /home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/lib/Basic/FileSystem.cpp:15:
/home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/include/llbuild/Basic/Stat.h:62:9: warning: 'S_ISLNK' macro redefined [-Wmacro-redefined]
#define S_ISLNK(mode) ((mode) & _S_IFMT) == S_IFLNK
        ^
/usr/include/x86_64-linux-gnu/sys/stat.h:136:10: note: previous definition is here
# define S_ISLNK(mode)   __S_ISTYPE((mode), __S_IFLNK)
         ^
/home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/lib/Basic/FileSystem.cpp:56:14: error: use of undeclared identifier '_S_IFMT'
    else if (S_ISLNK(Status.st_mode))
             ^
/home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/include/llbuild/Basic/Stat.h:62:33: note: expanded from macro 'S_ISLNK'
#define S_ISLNK(mode) ((mode) & _S_IFMT) == S_IFLNK
                                ^
1 warning and 1 error generated.
In file included from /home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/lib/Basic/FileInfo.cpp:15:
/home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/include/llbuild/Basic/Stat.h:62:9: warning: 'S_ISLNK' macro redefined [-Wmacro-redefined]
#define S_ISLNK(mode) ((mode) & _S_IFMT) == S_IFLNK
        ^
/usr/include/x86_64-linux-gnu/sys/stat.h:136:10: note: previous definition is here
# define S_ISLNK(mode)   __S_ISTYPE((mode), __S_IFLNK)
         ^
1 warning generated.
In file included from /home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/lib/Basic/PlatformUtility.cpp:11:
/home/buildnode/disk2/workspace/swift-llbuild-PR-Linux-smoke-test/llbuild/include/llbuild/Basic/Stat.h:62:9: warning: 'S_ISLNK' macro redefined [-Wmacro-redefined]
#define S_ISLNK(mode) ((mode) & _S_IFMT) == S_IFLNK
        ^
/usr/include/x86_64-linux-gnu/sys/stat.h:136:10: note: previous definition is here
# define S_ISLNK(mode)   __S_ISTYPE((mode), __S_IFLNK)
         ^
1 warning generated.
ninja: build stopped: subcommand failed.
```

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Comment by hughbe** [comment]
@swift-ci please smoke test

**Pull request merged**